### PR TITLE
remove unnecessary conditionals

### PR DESCRIPTION
**Pull request by dundargoc**
The reevaluate in optionstr is a bit suspect since I've almost removed
most of the if statement, not sure if I've missed something.

### CODE REVIEW

**Comment by brammool** [comment]
The change with "reevaluate" looks wrong, I'll skip it.  Still, no test fails, so perhaps we are missing a test for the %{ statusline item.

File: src/optionstr.c [hunk 1]
```diff
@@ -625,10 +625,8 @@ check_stl_option(char_u *s)
 	}
 	if (*s == '{')
 	{
-	    int reevaluate = (*s == '%');
```
Comment (zeertzjq) [inline src/optionstr.c:628]: I think there is bug that needs to be fixed.

Comment (zeertzjq) [inline src/optionstr.c:628]: I think there is bug that needs to be fixed.

**Comment by dundargoc** [comment]
Fair enough, although I will say that the original code still looks a bit fishy.

```c
	if (*s == '{')
	{
	    int reevaluate = (*s == '%');

	    s++;
	    while ((*s != '}' || (reevaluate && s[-1] != '%')) && *s)
			s++;
	    if (*s != '}')
			return N_(e_unclosed_expression_sequence);
	}
```

I don't understand how `reevaluate` isn't always equal to `FALSE` in this scenario.

**Comment by zeertzjq** [comment]
This causes a stack buffer overflow:
```
set statusline=%{%}
redrawstatus
```

**Comment by brammool** [comment]
> Fair enough, although I will say that the original code still looks a bit fishy.
> 
> ```c
> 	if (*s == '{')
> 	{
> 	    int reevaluate = (*s == '%');
> 
> 	    s++;
> 	    while ((*s != '}' || (reevaluate && s[-1] != '%')) && *s)
> 			s++;
> 	    if (*s != '}')
> 			return N_(e_unclosed_expression_sequence);
> 	}
> ```
> 
> I don't understand how `reevaluate` isn't always equal to `FALSE` in
> this scenario.

ou are right it should use s[1] == '%' instead of *s == '%'.  And
apparently a test with "{% ... }" is missing.  Can you make a PR
for that?

-- 
The budget process was invented by an alien race of sadistic beings who
resemble large cats.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Pull request closed without merging**